crypto/x509/x509_vfy.c

```diff
@@ -1851,6 +1851,31 @@ int X509_cmp_time(const ASN1_TIME *ctm, time_t *cmp_time)
     return ret;
 }
 
+/*
+ * Return 0 if time should not be checked or reference time is in range,
+ * or else 1 if it is past the end, or -1 if it is before the start
+ */
+int X509_cmp_timeframe(const X509_VERIFY_PARAM *vpm,
+                       const ASN1_TIME *start, const ASN1_TIME *end)
```

Comment:
I notice that the function operates on an `X509_VERIFY_PARAM`, and that makes me wonder whether this would be better called `X509_VERIFY_PARAM_cmp_timeframe`. Normally the initial part of the name "X509" implies the object that we operate on. If we rename it, then it is also probably better placed in the x509_vpm.c file. OTOH, I notice that other X509_* time functions don't conform to this, and also that vpm can be NULL, which is somewhat unusual.

So, I'd like other opinions on this.,

Reply:
Another option would be to change the order of the parameters, putting `vpm` last.
Yet since the semantics of the function is governed by the `vpm` parameter I appeared more intuitive to have it first, but others may see this differently.

Reply:
Heh, the name discussion goes on...

This function doesn't *operate on* a `X509_VERIFY_PARAM` (i.e. it's not a constructor, destructor, getter or setter), the first parameter is simply *input* just as much as the others.  This function is simply part of the `X509` API, and the way I see it, the current name is consistent with the rest.  `X509_VERIFY_PARAM_cmp_timeframe` would convey a whole different message, and actually be confusing about what it does.

Reply:
I could *possibly* see `X509_VERIFY_cmp_timeframe()`, but since we don't have a specific `X509_VERIFY` namespace, that simply doesn't fit.

Reply:
I agree that here the 'X509_VERIFY_PARAM *` parameter, though it us the first parameter, should not influence the name of the function.

Reply:
If we really wanna point out what the new function and its already existing friends operate on then all their names should start with `ASN1_TIME_`. Yet also this would be the topic of a different PR.